src/snowflake/cli/api/console/console.py

```diff
@@ -44,29 +44,38 @@ class CliConsole(AbstractConsole):
     """
 
     _indentation_level: int = INDENTATION_LEVEL
+    _extra_indent: int = 0
     _styles: dict = {
         "default": "",
         Output.PHASE: PHASE_STYLE,
         Output.STEP: STEP_STYLE,
-        Output.INFO: INFO_STYLE,
+        Output.INFO: None,
         Output.IMPORTANT: IMPORTANT_STYLE,
     }
 
     def _format_message(self, message: str, output: Output) -> Text:
         """Wraps message in rich Text object and applies formatting."""
         style = self._styles.get(output, "default")
-        text = Text(message, style=style)
+        if style is not None:
+            text = Text(message, style=style)
+        else:
+            text = Text.from_markup(message)
 
+        current_indent = self._extra_indent
         if self.in_phase and output in {Output.STEP, Output.INFO, Output.IMPORTANT}:
-            text.pad_left(self._indentation_level)
-
+            current_indent += 1
+        text.pad_left(current_indent * self._indentation_level)
         return text
 
     @contextmanager
     def phase(self, enter_message: str, exit_message: Optional[str] = None):
         """A context manager for organising steps into logical group."""
         if self.in_phase:
             raise CliConsoleNestingProhibitedError("Only one phase allowed at a time.")
+        if self._extra_indent > 0:
+            raise CliConsoleNestingProhibitedError(
+                "Phase cannot be used in an indented block."
+            )
```

Comment:
Conceptually, when would I use phase and when would I use indenting as a client of this API?

Reply:
phase vs step predates this PR, but conceptually, a step is a key part of a complex workflow. A phase is a group of related steps. A message is some additional information that's emitted during a step (at least that's how I think of it). Really everything other than phase or step is just a regular message. warning etc are just there for formatting. indented is the new part, it lets you indent messages/warnings. It shouldn't be used too often, but in cases like the diff output I'm adding here, it makes sense to visually separate the diff itself (via indentation) from the other messages within the phases/steps.

Reply:
It's not really about connections, e.g. bundle uses a phase for codegen but (other than telemetry) is a fully offline operation.